bitnami/janusgraph/templates/_helpers.tpl

```diff
@@ -0,0 +1,255 @@
+  resources: {{- toYaml .Values.volumePermissions.resources | nindent 12 }}
+  {{- else if ne .Values.volumePermissions.resourcesPreset "none" }}
+  resources: {{- include "common.resources.preset" (dict "type" .Values.volumePermissions.resourcesPreset) | nindent 12 }}
+  {{- end }}
+  volumeMounts:
+    - name: data
+      mountPath: {{ .Values.persistence.mountPath }}
+    - name: tmp-dir
+      mountPath: /tmp
+{{- end -}}
+
+{{/*
+Returns true if a storage backend has been configured
+*/}}
+{{- define "janusgraph.storage.enabled" -}}
+{{- if .Values.storageBackend.cassandra.enabled -}}
+{{- true -}}
+{{- else if .Values.storageBackend.berkeleyje.enabled -}}
+{{- true -}}
+{{- else if .Values.storageBackend.external.backend -}}
+{{- true -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the configured storage backend
+*/}}
+{{- define "janusgraph.storage.backend" -}}
+{{- if .Values.storageBackend.cassandra.enabled -}}
+{{- print "cql" -}}
+{{- else if .Values.storageBackend.berkeleyje.enabled -}}
+{{- print "berkeleyje" -}}
+{{- else if .Values.storageBackend.external.backend -}}
+{{- printf "%s" .Values.storageBackend.external.backend -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the hostname of the configured storage backend
+*/}}
+{{- define "janusgraph.storage.hostname" -}}
+{{- if .Values.storageBackend.cassandra.enabled -}}
+{{- include "common.names.dependency.fullname" (dict "chartName" "cassandra" "chartValues" .Values.storageBackend.cassandra "context" $) -}}
+{{- else if .Values.storageBackend.external.hostname -}}
+{{- printf "%s" .Values.storageBackend.external.hostname -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the port of the configured storage backend
+*/}}
+{{- define "janusgraph.storage.port" -}}
+{{- if .Values.storageBackend.cassandra.enabled }}
+{{- printf "%d" (int .Values.cassandra.service.ports.cql) -}}
+{{- else if .Values.storageBackend.external.port -}}
+{{- printf "%d" (int .Values.storageBackend.external.port) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create the storage password secret name
+*/}}
+{{- define "janusgraph.storage.username" -}}
+    {{- if .Values.storageBackend.cassandra.enabled -}}
+        {{- printf "%s" (default "cassandra" .Values.cassandra.dbUser.user) -}}
+    {{- else if .Values.storageBackend.external.username -}}
+        {{- .Values.storageBackend.external.username -}}
+    {{- end -}}
+{{- end -}}
+
+{{/*
+Create the storage password secret name
+*/}}
+{{- define "janusgraph.storage.password.secretName" -}}
+    {{- if .Values.storageBackend.cassandra.enabled -}}
+        {{- printf "%s-cassandra" (include "common.names.fullname" .) -}}
+    {{- else if .Values.storageBackend.external.existingSecret -}}
+        {{- .Values.storageBackend.external.existingSecret -}}
+    {{- end -}}
+{{- end -}}
+
+{{/*
+Create the storage password secret key
+*/}}
+{{- define "janusgraph.storage.password.secretKey" -}}
+    {{- if .Values.storageBackend.cassandra.enabled -}}
+        cassandra-password
```

Comment:
Aren't these spaces problematic?

Reply:
Surprisingly not, but  I removed them to be aligned with the rest of the code